.github/workflows/update-types.yml

```diff
@@ -88,4 +95,39 @@ jobs:
             Update types from ${{ steps.get_swagger_gh_uri.outputs.gh_uri }} ${{ github.event.inputs.single_path && format('(single path: {0})', github.event.inputs.single_path) || '' }}
           branch: ${{ github.event.inputs.single_path && format('autogenerate-{0}', github.event.inputs.single_path) || 'autogenerate' }}
           push_options: '--force'
-          create_branch: true
\ No newline at end of file
+          create_branch: true
+
+      - name: Log in to Azure
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.PUBLISH_CLIENT_ID }}
+          tenant-id: ${{ secrets.PUBLISH_TENANT_ID }}
+          subscription-id: ${{ secrets.PUBLISH_SUBSCRIPTION_ID }}
+
+      - name: Tar+Gzip all json files under the generated folder
+        run: find . -name '*.json' -print0 | tar -czf ../types.tgz --null --files-from -
+        working-directory: generated
+
+      - name: List all files in the generated directory
+        run: ls -la
+
+      - name: Get current date
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
```

Comment:
This works great for a prototype, but relying on shell scripts in a pipeline feels problematic for maintainability in the long run (difficulty in testing, safely making changes). Have you thought about how you're going to implement it "for real"?

IMO it might be worth chatting to @shenglol about the tool he created to publish BMR modules: https://github.com/Azure/bicep/blob/main/src/Bicep.RegistryModuleTool/Program.cs.

Reply:
I disagree that its problematic, consider the fact that ORAS is a OSS product that is tested and mantained by the ACR team so I think its ok to rely on it. I am open to talk more about it though, can you articulate more specifically what are the concerns?